LCM/scripts/RemoveModule.py

```diff
@@ -51,40 +51,22 @@ def RemoveMirroredDirectory(moduleDir, omiDir):
 print("Removing module " + moduleName)
 
 # special section for nxOMSAutomationWorker module
-# the Linux Hybrid worker needs to be killed before the module is removed
-# also a good idea to remove the state.conf file
-# manually invoking RemoveModule.py will preserve worker.conf file in case the user wants to keep their settings
-# an uninstall or purge operation of the OMS agent installer will get rid of worker.conf file
+# the Linux Hybrid worker and manager needs to be killed before the module is removed
+# also a good idea to remove the state and working directory
 if moduleName == "nxOMSAutomationWorker":
-    registration_file_path = "/opt/microsoft/omsconfig/modules/nxOMSAutomationWorker/DSCResources/MSFT_nxOMSAutomationWorkerResource/automationworker/scripts/register_oms.py"
-    state_conf_location = "/var/opt/microsoft/omsagent/state/automationworker/state.conf"
-    config_state_section = "state"
-    config_pid_key = "pid"
-    workspace_id_key = "workspace_id"
+    state_dir_path = "/var/opt/microsoft/omsagent/state/automationworker"
+    working_dir_path = "/var/opt/microsoft/omsagent/run/automationworker"
     automation_user = "nxautomation"
-    try:
-        import ConfigParser
-    except ImportError:
-        # ConfigParser is named configparser in python 3x
-        import configparser as ConfigParser
 
     # invoke deregister when it becomes available
+    # Kill all processes running as nxautomation
+    subprocess.call(["sudo", "pkill", "-u", automation_user, ".*"])
```

Comment:
Depending on the whether the solution was enabled or not, there could be cases where the return code could be non-zero, but we have no error. e.g when the resource wasn't enabled. Since pkill finds nothing to kill, it returns non-zero code. But no resource was not supposed to be running in the first place. To determine if the non-zero return code actually means an error, we'll need access to the DSC mof file at RemoveModule.py level.

Hence, I leave it as a best effort pkill attempt.